samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -74,7 +74,7 @@ public GroupByContainerCount(int containerCount) {
     // Convert to a Set of ContainerModel
     Set<ContainerModel> containerModels = new HashSet<>();
     for (int i = 0; i < containerCount; i++) {
-      containerModels.add(new ContainerModel(i, taskGroups[i]));
+      containerModels.add(new ContainerModel(String.valueOf(i), taskGroups[i]));
```

Comment:
It "generates" some containerIds according to the count. It receives others from external sources like the task assignment manager. I was only suggesting that the latter should be validated s.t. we throw an exception rather than letting unexpected/undefined behavior occur.